Marlin/src/HAL/DUE/HAL.h

```diff
@@ -38,33 +38,33 @@
 
 #include "../../core/serial_hook.h"
 
-typedef ForwardSerial0Type< decltype(Serial) > DefaultSerial;
-typedef ForwardSerial0Type< decltype(Serial1) > DefaultSerial1;
-typedef ForwardSerial0Type< decltype(Serial2) > DefaultSerial2;
-typedef ForwardSerial0Type< decltype(Serial3) > DefaultSerial3;
-extern DefaultSerial MSerial;
-extern DefaultSerial1 MSerial1;
-extern DefaultSerial2 MSerial2;
-extern DefaultSerial3 MSerial3;
+typedef ForwardSerial1Class< decltype(Serial) > DefaultSerial1;
+typedef ForwardSerial1Class< decltype(Serial1) > DefaultSerial2;
+typedef ForwardSerial1Class< decltype(Serial2) > DefaultSerial3;
+typedef ForwardSerial1Class< decltype(Serial3) > DefaultSerial4;
+extern DefaultSerial1 MSerial;
```

Comment:
While fixing a bug with teensy, I was wondering: why not call those MSerial0 directly ? So that we can preprocessor them with the serial port: 
```
#define DECLARE_SERIAL(X) typedef XXX<decltype(Serial##X)> DefaultSerial##(INCREMENT(X)); extern DefaultSerial##INCREMENT(X) MSerial##X

// In effect, let's have a MSerial0, MSerial1 and so on.
DECLARE_SERIAL(SERIAL_PORT)
```
Let's drop Arduino dumbness about the first serial being different so that we always have prefix + index here.

Reply:
Where we have control over the name, we can certainly add the zero. I was superstitious that such a simple name must come from a platform, but now I think the M stands for "Marlin."

Reply:
Meanwhile, the Arduino-sketchy names "`MYSERIAL0/1`" are seeming kind of quaint now. Any of `SERIAL(n)` or `MSERIAL(n)` or maybe `MFSERIAL(n)` would be better.

Reply:
Even a bit more hardcode, but what about `#define Serial0 Serial` and be done with special case for 0 ?

Reply:
There are so many *X* Serial *Y* in the code base now, my head hurts...

Reply:
If the composition is working as expected, there should be only one SERIAL_IMPL in the code base (except for HAL), the rest should be hidden underneath the carpet. So let's simplify all special cases here so we can preprocess as much as we can and remove all special case that are not preprocessor friendly, IMHO.